### PR TITLE
[MRG] FIX: do not convert to BrainVision if non-Volt channels are present

### LINKED ISSUE #618: pybv for BV conversion: raw MUST drop all non Volt channels
We are currently simply making a raw.get_data() call and write the whole data using pybv, see:

https://github.com/mne-tools/mne-bids/blob/e2755da1136aa7eb473f0b05d7efc568b66a1e0a/mne_bids/write.py#L707

This is problematic, because pybv's assumption is that **all data channels are in Volts**.

This is not necessarily the case for an MNE raw object, so this can lead to bugs, where e.g. an aux channel that was in degrees Celsius, is suddenly encoded in Volts and does not make sense anymore.

Can be fixed by doing a `raw.pick_types` for all types that use Volts, prior to `raw.get_data()`.

Obviously, we'd need to warn the user if types are dropped.

### PR DESCRIPTION
**Pull request by sappelhoff**
closes #618 

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [x] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: mne_bids/write.py [hunk 1]
```diff
@@ -693,6 +694,21 @@ def _write_raw_brainvision(raw, bids_fname, events):
...
+    chtypes_volt = {chtype: True for chtype, unit in chtype_units.items()
+                    if unit[-1] == 'V'}
+    ch_idxs = mne.pick_types(raw.info, meg=False, **chtypes_volt)
+    if len(ch_idxs) != len(raw.ch_names):
+        non_volt_chs = set(raw.ch_names) - set(np.array(raw.ch_names)[ch_idxs])
+        msg = ('Conversion to BrainVision format needed to be stopped, '
+               'because your raw data contains channel types that are '
+               f'not represented in Volts: "{non_volt_chs}"'
+               '\n\nUntil BrainVision format conversion is improved, you '
+               'can drop these channels from your raw data before using '
```
Comment (hoechenberger) [inline mne_bids/write.py:708]:
```suggestion
               'must drop these channels from your raw data before using '
```

**Comment by sappelhoff** [comment]
Mmh getting an error here that looks like an MNE-Python error:

> TypeError: pick_types() got an unexpected keyword argument 'fnirs_cw_amplitude'

fnirs_cw_amplitude is listed in the "units" defaults in MNE:

https://github.com/mne-tools/mne-python/blob/28b36c06a744074b54993128fa1908ea10464ff2/mne/defaults.py#L15-L18

so either:

1. I am mistaken in assuming that keys in the units dict are channel types, or
2. the pick_channels etc functions must be updated to contain the new type "fnirs_cw_amplitude"

cc @larsoner

**Comment by sappelhoff** [comment]
> I am mistaken in assuming that keys in the units dict are channel types, or

:point_up: this is the case. It seems like the channel type `"fnirs"` has some subtypes, of which "fnirs_cw_amplitude" and "fnirs_od" are two. Both of these are measured in Volts, but I am going to exclude them here --> fNIRS data should be written to [snirf](https://github.com/fNIRS/snirf), not BrainVision.

**Comment by sappelhoff** [comment]
This part of the test is failing:

https://github.com/mne-tools/mne-bids/blob/e2755da1136aa7eb473f0b05d7efc568b66a1e0a/mne_bids/tests/test_write.py#L330-L365

But I don't really get it. At the end of the code snippet I pasted here, `raw` and `raw2` get compared ... but raw2 is the same as raw, just with picked ch_types --> because pick_types works in place, so this one is weird: `raw2 = raw.pick_types(meg=False, eeg=True, stim=True, eog=True, ecg=True)`

**Comment by sappelhoff** [comment]
> But I don't really get it. At the end of the code snippet I pasted here, raw and raw2 get compared ... but raw2 is the same as raw, just with picked ch_types --> because pick_types works in place, so this one is weird:

Let me rephrase my question: What do we want with that test, and how can we achieve it in a clear way?

https://github.com/mne-tools/mne-bids/blob/e2755da1136aa7eb473f0b05d7efc568b66a1e0a/mne_bids/tests/test_write.py#L330-L365

**Review by hoechenberger — APPROVED** [review]

**Comment by hoechenberger** [comment]
@sappelhoff We really should rework most of the tests someday, they're convoluted and full of potential side-effects :(

**Comment by jasmainak** [comment]
what needs to be done to support non-volt channels in `pybv` ? glad to see these issues help development of `pybv`

**Comment by sappelhoff** [comment]
> what needs to be done to support non-volt channels in pybv

I think we need to solve this one: https://github.com/bids-standard/pybv/issues/50

in short: 
- currently we can pass individual "resolutions" per channel
- now we need the ability to also pass indiviual "units" per channel

it should be relatively straight forward, if you wanna give it a try :wink:

**Comment by sappelhoff** [comment]
@hoechenberger ready for merge?

File: doc/whats_new.rst [hunk 2]
```diff
@@ -45,6 +45,7 @@ Bug fixes
 - Fix a bug in :func:`mne_bids.write_raw_bids`, where passing raw data with :class:`mne.Annotations` set and the ``event_id`` dictionary not containing the :class:`mne.Annotations` descriptions as keys would raise an error, by `Richard Höchenberger`_ (`#603 <https://github.com/mne-tools/mne-bids/pull/603>`_)
 - Fix a bug in :func:`mne_bids.write_raw_bids` when passing raw MEG data with Internal Active Shielding (IAS) from Triux system, by `Alexandre Gramfort`_ (`#616 <https://github.com/mne-tools/mne-bids/pull/616>`_)
 - Fix a bug in :func:`mne_bids.write_raw_bids`, where original format of data was not kept when writing to FIFF, by `Alexandre Gramfort`_, `Stefan Appelhoff`_, and `Richard Höchenberger`_ (`#610 <https://github.com/mne-tools/mne-bids/pull/610>`_)
+- Fix a bug where conversion to BrainVision format was done even when non-Volt channel types were present in the data (BrainVision conversion is done by pybv, which currently only supports Volt channel types), by `Stefan Appelhoff`_ (`#619 <https://github.com/mne-tools/mne-bids/pull/619>`_)
```
Comment (hoechenberger) [inline doc/whats_new.rst:48]:
```suggestion
- Fix a bug where conversion to BrainVision format was done even when non-Volt channel types were present in the data (BrainVision conversion is done by `pybv`, which currently only supports Volt channel types), by `Stefan Appelhoff`_ (`#619 <https://github.com/mne-tools/mne-bids/pull/619>`_)
```

**Review by hoechenberger — APPROVED** [review]
Small nitpick, otherwise LGTM!

**Review by hoechenberger — APPROVED** [review]

**Comment by hoechenberger** [comment]
My suggestion broke Circle… will fix

File: doc/whats_new.rst [hunk 3]
```diff
@@ -45,6 +45,7 @@ Bug fixes
 - Fix a bug in :func:`mne_bids.write_raw_bids`, where passing raw data with :class:`mne.Annotations` set and the ``event_id`` dictionary not containing the :class:`mne.Annotations` descriptions as keys would raise an error, by `Richard Höchenberger`_ (`#603 <https://github.com/mne-tools/mne-bids/pull/603>`_)
 - Fix a bug in :func:`mne_bids.write_raw_bids` when passing raw MEG data with Internal Active Shielding (IAS) from Triux system, by `Alexandre Gramfort`_ (`#616 <https://github.com/mne-tools/mne-bids/pull/616>`_)
 - Fix a bug in :func:`mne_bids.write_raw_bids`, where original format of data was not kept when writing to FIFF, by `Alexandre Gramfort`_, `Stefan Appelhoff`_, and `Richard Höchenberger`_ (`#610 <https://github.com/mne-tools/mne-bids/pull/610>`_)
+- Fix a bug where conversion to BrainVision format was done even when non-Volt channel types were present in the data (BrainVision conversion is done by `pybv`, which currently only supports Volt channel types), by `Stefan Appelhoff`_ (`#619 <https://github.com/mne-tools/mne-bids/pull/619>`_)
```
Comment (hoechenberger) [inline doc/whats_new.rst:48]:
```suggestion
- Fix a bug where conversion to BrainVision format was done even when non-Volt channel types were present in the data (BrainVision conversion is done by ``pybv``, which currently only supports Volt channel types), by `Stefan Appelhoff`_ (`#619 <https://github.com/mne-tools/mne-bids/pull/619>`_)
```

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
thx @sappelhoff
